src/Cpp/1-getting-started/1-HelloWindow/Application.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <string_view>
+
+// ReSharper disable once CppInconsistentNaming
+struct GLFWwindow;
+class Input;
+
+class Application
+{
+public:
+    Application(const std::string_view title);
+    virtual ~Application();
+    void Run();
+
+protected:
+    virtual void Cleanup();
+    void Close();
+    virtual bool Initialize();
+    [[nodiscard]] bool IsButtonPressed(const std::int32_t button) const;
+    [[nodiscard]] bool IsKeyDown(const std::int32_t key) const;
+    [[nodiscard]] bool IsKeyPressed(const std::int32_t key) const;
+    [[nodiscard]] bool IsKeyUp(const std::int32_t key) const;
+    virtual void Render() = 0;
+    virtual void Update() = 0;
+
+    std::unique_ptr<Input> _input;
+
+private:
+    GLFWwindow* _window;
+    std::int32_t _width;
```

Comment:
prefer direct member initialisation? `GLFWwindow* _window = nullptr;`

Reply:
We solved it by vote see [here](https://github.com/GraphicsProgramming/learnd3d11/discussions/17)